spec/unit/puppet/type/grafana_datasource_type_spec.rb

```diff
@@ -58,7 +58,7 @@
       expect(gdatasource[:name]).to eq('foo')
       expect(gdatasource[:grafana_url]).to eq('http://example.com')
       expect(gdatasource[:url]).to eq('http://es.example.com')
-      expect(gdatasource[:type]).to eq(:elasticsearch)
+      expect(gdatasource[:type]).to eq('elasticsearch')
```

Comment:
mhm, that changes it from a symbol to a string. I'm not sure if that has any other impact. Instead you could expand the list of available values in the grafana_datasource.rb?

Reply:
With the `newvalues()` from the Type removed, there's no conversion from string to Ruby symbol, so the data passed into the Type is now a string (I just changed the test to make the test pass).  If you think there's no value checking I can remove it.  As for adding known values, there's quite a lot already:

https://grafana.com/plugins?type=datasource

and adding that list doesn't cover people using unreleased grafana data sources with this Type.

Reply:
Thanks for the explanation!